src/plugins/unified_histogram/public/chart/use_request_params.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { connectToQueryState, QueryState } from '@kbn/data-plugin/public';
+import { createStateContainer, useContainerState } from '@kbn/kibana-utils-plugin/public';
+import { useEffect, useMemo, useRef } from 'react';
+import type { UnifiedHistogramRequestContext, UnifiedHistogramServices } from '../types';
+
+export const useRequestParams = ({
```

Comment:
I originally thought it was a good idea to have unified histogram track changes to the query, filters, and time range itself, but in hindsight this should be the consumer's responsibility. I'll change this to get these as props in a followup PR to prepare for solutions usage.